app/scripts/modules/core/src/managed/plugins/index.ts

```diff
@@ -0,0 +1,12 @@
+import { constraintsManager, IConstraintHandler } from '../constraints/registry';
+import { IResourceKindConfig, resourceManager } from '../resources/resourceRegistry';
+
+export interface IManagedDeliveryPlugin {
+  resources?: IResourceKindConfig[];
+  constraints?: IConstraintHandler[];
+}
+
+export const registerPlugin = (plugin: IManagedDeliveryPlugin) => {
```

Comment:
These end up in the top level export, so `registerPlugin` would be a misleading name
```suggestion
export const registerManagedDeliveryPlugin = (plugin: IManagedDeliveryPlugin) => {
```